backend/src/handlers/BaseHandler.ts

```diff
@@ -23,18 +25,43 @@ export abstract class BaseHandler<TPayload, TOutput extends BaseOutput> implemen
   /**
    *  @param validationSchema - The Joi schema that the payload will be validated against
    */
-  constructor(private validationSchema: Joi.ObjectSchema<TPayload>) {}
+  constructor(private validationSchema: Joi.ObjectSchema<TPayload>, private enumType: { [key: string]: string }) {}
 
   protected abstract getResult(payload: Result<TPayload>, res: Response<TOutput>): any;
 
   public runHandler(req: Request<TPayload>, res: Response<BaseOutput>) {
     const payload = validatePayload<TPayload>(this.validationSchema, req.body);
 
     if (payload.isError()) {
-      res.status(400).send({ ok: false, error: payload.getError() });
+      const error = payload.getError();
+      const apiError = mapBaseErrorToAPIError(error);
+
+      res.status(400).send({ ok: false, error: apiError });
       return;
     }
 
     return this.getResult(payload, res);
   }
+
+  protected handleError(error: IBaseError, res: Response) {
+    return ErrorHandler.handleError(error, this.enumType, res);
+  }
+}
+
+// TODO: Move this to a separate file and tidy up / test
+class ErrorHandler {
+  static isValidErrorCode(errorCode: string, enumType: { [key: string]: string }): boolean {
+    return Object.values(enumType).includes(errorCode);
+  }
+
+  static handleError(error: IBaseError, enumType: { [key: string]: string }, res: Response) {
+    if (this.isValidErrorCode(error.code, enumType)) {
+      return res.send({
+        ok: false,
+        error: mapBaseErrorToAPIError(error),
+      });
+    }
+
+    throw error;
+  }
```

Comment:
`handleError` checks the appropriate enum for the errorCode and if there is a match, it returns the error mapping it to the APIError type. 

This ensures we don't send back errors that the client shouldn't see (internal knowledge). 

The `mapBaseErrorToAPIError` simply changes the shape, so you can't call `res` with an error unless you have explicitly made it this type or used this function. In otherwords it makes it very hard to send back an error we don't want the client to see, but if you have defined it in the enum and use `this.handleError` you should be good (and it's simple)